mesheryctl/cmd/start.go

```diff
@@ -42,33 +45,75 @@ var startCmd = &cobra.Command{
 			log.Fatal(err)
 		}
 
-		fmt.Println("Starting Meshery...")
+		log.Info("Starting Meshery...")
 		start := exec.Command("docker-compose", "-f", dockerComposeFile, "up", "-d")
 		start.Stdout = &out
 		start.Stderr = &stderr
 
 		if err := start.Run(); err != nil {
-			fmt.Println(stderr.String())
+			log.Println(stderr.String())
 			return
 		}
 
-		fmt.Println("Opening Meshery in your broswer. If Meshery does not open, please point your browser to http://localhost:9081 to access Meshery.")
-		ostype, err := exec.Command("uname", "-s").Output()
+		checkFlag := 0 //flag to check
+
+		//connection to docker-client
+		cli, err := client.NewEnvClient()
 		if err != nil {
-			log.Fatal("[WARNING] Unable to detect OS type. Warning message: \n", err)
+			panic(err)
 		}
-		os := strings.TrimSpace(string(ostype))
 
-		// Link to Meshery User Interface
-		url := "http://localhost:9081"
+		containers, err := cli.ContainerList(context.Background(), types.ContainerListOptions{})
+		if err != nil {
+			panic(err)
+		}
 
-		if os == "Linux" {
-			// Meshery running on Linux host
-			exec.Command("xdg-open", url).Start()
-		} else {
+		//check for container meshery_meshery_1 running status
+		for _, container := range containers {
+			if "/meshery_meshery_1" == container.Names[0] {
+				log.Info("Opening Meshery in your broswer. If Meshery does not open, please point your browser to http://localhost:9081 to access Meshery.")
+				ostype, err := exec.Command("uname", "-s").Output()
+				if err != nil {
+					log.Fatal("[WARNING] Unable to detect OS type. Warning message: \n", err)
+				}
+				os := strings.TrimSpace(string(ostype))
+				// Link to Meshery User Interface
+				url := "http://localhost:9081"
+				if os == "Linux" {
+					// Meshery running on Linux host
+					exec.Command("xdg-open", url).Start()
+				} else {
+					// Asssume Meshery running on MacOS host
+					exec.Command("open", url).Start()
+				}
+				checkFlag = 0
+				break
+			} else {
+				checkFlag = 1
+			}
+		}
 
-			// Asssume Meshery running on MacOS host
-			exec.Command("open", url).Start()
+		//if meshery_meshery_1 failed to start showing logs
+		//code for logs
+		if checkFlag == 1 {
+			log.Info("Starting Meshery logging . . .")
+			cmdlog := exec.Command("docker-compose", "-f", dockerComposeFile, "logs", "-f")
```

Comment:
we don't need a "-f" right after docker-compose, I think.

Reply:
I built mesheryctl after removing -f for the command, and it's not wokring. 

INFO[0001] Starting Meshery...                          
ERRO[0002] No such command: .meshery/meshery.yaml

Commands:
  build              Build or rebuild services
  bundle             Generate a Docker bundle from the Compose file
  config             Validate and view the 
...
and other docker-compose command.
So I think  going forward with -f would be better.